src/toolkit-cleaner/index.ts

```diff
@@ -0,0 +1,108 @@
+import * as path from 'path';
+import { Duration } from 'aws-cdk-lib';
+import { DockerImageAsset } from 'aws-cdk-lib/aws-ecr-assets';
+import { Rule, Schedule } from 'aws-cdk-lib/aws-events';
+import { SfnStateMachine } from 'aws-cdk-lib/aws-events-targets';
+import { PolicyStatement } from 'aws-cdk-lib/aws-iam';
+import { NodejsFunction } from 'aws-cdk-lib/aws-lambda-nodejs';
+import { Asset } from 'aws-cdk-lib/aws-s3-assets';
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
+import * as tasks from 'aws-cdk-lib/aws-stepfunctions-tasks';
+import { Construct } from 'constructs';
+
+/**
+ * Properties for a ToolkitCleaner
+ */
+export interface ToolkitCleanerProps {
+  /**
+   * The schedule for the cleaner
+   *
+   * @default - every day
+   */
+  readonly schedule?: Schedule;
+}
+
+/**
+ * Clean unused S3 and ECR assets from your CDK Toolkit.
+ */
+export class ToolkitCleaner extends Construct {
+  constructor(scope: Construct, id: string, props: ToolkitCleanerProps = {}) {
+    super(scope, id);
+
+    // Dummy assets to reference S3 bucket and ECR repository
+    const fileAsset = new Asset(this, 'FileAsset', {
+      path: path.join(__dirname, '..', '..', 'assets', 'toolkit-cleaner', 'docker', 'dummy.txt'),
+    });
+    const dockerImageAsset = new DockerImageAsset(this, 'DockerImageAsset', {
+      directory: path.join(__dirname, '..', '..', 'assets', 'toolkit-cleaner', 'docker'),
+    });
+
+    const getStackNamesHandler = new NodejsFunction(this, 'get-stack-names');
```

Comment:
Curious - why are you not using projen's lambda support to handle pre-bundling...

Reply:
Because I forgot about it :)